### PR TITLE
Add more logging and fixing some bugs

### PR DESCRIPTION
**Pull request by jcagme**
Adding more logging so it is easier to find the dropped subscription triggers and at the same time fixing some bugs that (some) might (or might not) be related to that.

### CODE REVIEW

File: src/Maestro/DependencyUpdater/DependencyUpdater.cs [hunk 1]
```diff
@@ -168,12 +168,12 @@ select new
 
                 var subscriptionsAndBuilds = await subscriptionsToUpdate.ToListAsync(cancellationToken);
                 Logger.LogInformation($"Will update '{subscriptionsAndBuilds.Count}' subscriptions");
+
                 foreach (var s in subscriptionsAndBuilds)
                 {
                     Logger.LogInformation($"Will update {s.subscription} to build {s.latestBuild}");
+                    await UpdateSubscriptionAsync(s.subscription, s.latestBuild);
```
Comment (alexperovich) [inline src/Maestro/DependencyUpdater/DependencyUpdater.cs]: This should have a try/catch so one failure doesn't cause all the following subscriptions to be skipped.

Comment (mmitche) [reply src/Maestro/DependencyUpdater/DependencyUpdater.cs]: @alexperovich I think this actually ends up in ActionRunner, which has a try catch.

Comment (alexperovich) [reply src/Maestro/DependencyUpdater/DependencyUpdater.cs]: I see it does, the only exceptions that can get here are a problem with Logger.BeginScope, its probably fine then.

File: src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs [hunk 2]
```diff
@@ -438,13 +438,13 @@ public async Task<List<GitFile>> GetFilesAtCommitAsync(string repoUri, string co
             string requestUri,
             ILogger logger,
             string body = null,
-            string versionOverride = null)
+            string versionOverride = null,
+            int retryCount = 15)
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs]: Isn't 15 quite high?

**Pull request merged**